core/src/io/anuke/mindustry/world/blocks/distribution/Junction.java

```diff
@@ -30,6 +34,58 @@ public Junction(String name){
         entityType = JunctionEntity::new;
     }
 
+    @Override
+    public void load(){
+        super.load();
+
+        overlayArrowRegion = Core.atlas.find("overlay-arrow");
+        overlayArrowBodyRegion = Core.atlas.find("overlay-arrow-body");
+        overlayArrowShadeRegion = Core.atlas.find("overlay-arrow-shade");
+    }
+
+    @Override
+    public void drawDebug(Tile tile){
+        super.drawDebug(tile);
+
+        // Calculate proximity tiles
+        int proximityTiles = 0;
+        int proximityFade = 0;
+        for (Tile proximityTile : tile.entity.proximity()) {
+            byte relativePosition = tile.relativeTo(proximityTile);
+            if (proximityTile.block().outputsItems()) {
```

Comment:
`for(Tile proximityTile : tile.entity.proximity()){`
&
`if(proximityTile.block().outputsItems()){`